src/units/unit.cpp

```diff
@@ -2120,9 +2222,12 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 	} else if(apply_to == "remove_ability") {
 		if(const config& ab_effect = effect.child("abilities")) {
 			for(const config::any_child ab : ab_effect.all_children_range()) {
-				remove_ability_by_id(ab.cfg["id"]);
+				remove_ability_by_attribute(ab.cfg, false);
 			}
 		}
+		if(const config& fab_effect = effect.child("filter_abilities")) {
```

Comment:
I'm wondering if the tag should be `[filter_ability]` rather than `[filter_abilities]`, to match the singular/plural of `[filter_attack]`, `[filter_location]`, `[filter_side]`, etc. However, that wouldn't agree with the singular/plural of the existing `[effect]apply_to=remove_ability[abilities]` tag.

Reply:
[filter_abilities]could filter more what one ability, i willl let the actual name.

Reply:
changed to [filter_ability]